pkg/volume/csi/csi_plugin.go

```diff
@@ -217,11 +215,32 @@ func (p *csiPlugin) Init(host volume.VolumeHost) error {
 		if csiClient == nil {
 			klog.Warning(log("kubeclient not set, assuming standalone kubelet"))
 		} else {
-			// Start informer for CSIDrivers.
-			factory := csiapiinformer.NewSharedInformerFactory(csiClient, csiResyncPeriod)
-			p.csiDriverInformer = factory.Storage().V1beta1().CSIDrivers()
-			p.csiDriverLister = p.csiDriverInformer.Lister()
-			go factory.Start(wait.NeverStop)
+			// set CSIDriverLister
+			adcHost, ok := host.(volume.AttachDetachVolumeHost)
+			if ok {
+				p.csiDriverLister = adcHost.CSIDriverLister()
+				if p.csiDriverLister == nil {
+					klog.Error(log("CSIDriverLister not found on AttachDetachVolumeHost"))
+				}
+			}
+			kHost, ok := host.(volume.KubeletVolumeHost)
+			if ok {
+				csiDriversSynced := kHost.CSIDriversSynced()
+				if csiDriversSynced == nil {
+					klog.Error(log("CSIDriversSynced not found KubeletVolumeHost"))
+				} else {
+					synced := []cache.InformerSynced{}
+					synced = append(synced, csiDriversSynced)
+					if !cache.WaitForCacheSync(wait.NeverStop, synced...) {
```

Comment:
Instead of putting it in the Init(), I mean add this check to right before we actually need to use it in skipAttach and pod info on mount.

Reply:
Okay.  Will do.